yarn-project/aztec.js/src/contract/contract_function_interaction.ts

```diff
@@ -56,10 +84,10 @@ export class ContractFunctionInteraction extends BaseContractInteraction {
    * @param options - An optional object containing additional configuration for the transaction.
    * @returns An execution request wrapped in promise.
    */
-  public request(): FunctionCall {
+  public request(options: ExecutionRequestOptions = {}): FunctionCall {
     const args = encodeArguments(this.functionDao, this.args);
     const functionData = FunctionData.fromAbi(this.functionDao);
-    return { args, functionData, to: this.contractAddress };
+    return { args, functionData, to: this.contractAddress, static: options.static ?? false };
```

Comment:
Sorry I didn't think about it too much when you mentioned changing the aztec.js. But I think we probably don't need to provide the option to send a tx that has no state changes.

Reply:
I added it since it exists in other blockchain libs (such as `ethers.js`) and it felt weird to be able to statically call a contract from another contract, but not initiate a static context from an entrypoint. 

Also think it's useful as a dev to absolutely *ensure* I'm not modifying state when I don't want to and get an error if something has unintended side effects.

Reply:
It's common to static call a contract from another contract, for getting a value or checking some status while not causing any unintended side effects.

But it's not likely that someone will want to send a transaction that does not change any state at all. If they want to learn about some values in the contracts, they can use the view api. The callStatic api from ethers is more for simulating a tx and see if it will fail or not. In this case, we should add an estimateGas api instead of pretending a tx is a static call.

Reply:
Alright, understood! Should I leave it for the `simulate()` call?

Reply:
Removed from `aztec.js entirely!